### PR TITLE
Add pattern on writing greppable code

### PR DESCRIPTION
**Pull request by bockets**

### CODE REVIEW

File: _patterns/greppable-code.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+categories: Ruby, JavaScript, PHP, CSS, Rails
+name: Write Greppable Code
+---
+
+When writing code, strive to ensure that _named things_ can quickly be found by their name. Avoid using interpolation, clever pattern matching, or string assembly to translate names because this makes them harder to find. Be explicit, don't invent names based on their context.
+
+# Why?
+
+Other teams in the business know only the name of the thing thing, not the implementation. Therefor, using their name means that the implementation is self-documenting, and other engineers can quickly use that name to find where it is used.
```
Comment (baerbradford) [inline _patterns/greppable-code.md:10]:
```suggestion
Other teams in the business know only the name of the thing thing, not the implementation. Therefore, using their name means that the implementation is self-documenting, and other engineers can quickly use that name to find where it is used.
```

**Review by baerbradford — APPROVED** [review]
While I'm a little sad this explicitly calls out some of the recent tracking work as a pattern not to follow, it's a perfectly clear example demonstrating the issue. I do agree this is a better way to make sure we have a readable, searchable codebase.

**Comment by bockets** [comment]
> While I'm a little sad this explicitly calls out some of the recent tracking work as a pattern not to follow, it's a perfectly clear example demonstrating the issue. I do agree this is a better way to make sure we have a readable, searchable codebase.

I'm open to other examples! That was top of mind when I wrote this, but it's definitely not a problem unique to the recent tracking work. Would you prefer that I rewrote that?

**Comment by baerbradford** [comment]
> I'm open to other examples! That was top of mind when I wrote this, but it's definitely not a problem unique to the recent tracking work. Would you prefer that I rewrote that?

Nope, I don't think it needs to be rewritten. It's a perfect example. It quite literally confused me when the approach was first implemented for tracking because I tried searching for a string exactly like this. I was just ready to live with it once I understood what was going on 😆, but we probably aren't being too kind to new engineers or our future selves ignoring this pattern.

**Review by DyegoCosta — APPROVED** [review]

**Review by sogamoso — APPROVED** [review]

**Review by Hafiz-ahmed-dev — APPROVED** [review]

**Review by mgallagher0 — APPROVED** [review]

**Review by BPMattBauer — APPROVED** [review]

**Review by karlentwistle — APPROVED** [review]

**Review by bhanubhakta — APPROVED** [review]

**Pull request merged**